core/controllers/admin.py

```diff
@@ -639,14 +639,16 @@ class SendDummyMailToAdminHandler(base.BaseHandler):
 
     @acl_decorators.can_access_admin_page
     def post(self):
+        username = self.payload.get('username')
         if feconf.CAN_SEND_EMAILS:
             response = {
                 'msg': 'Success! Mail sent to admin.',
             }
-            dummy_email_name = 'DUMMY_SYSTEM_NAME'
-            dummy_email_address = 'dummy@system.com'
+            dummy_email_name = feconf.SYSTEM_EMAIL_NAME
+            dummy_email_address = feconf.SYSTEM_EMAIL_ADDRESS
+            email_body = 'This is a test mail from ' + username
             email_manager.send_dummy_mail_to_admin(
-                dummy_email_name, dummy_email_address)
+                dummy_email_name, dummy_email_address, email_body)
```

Comment:
You don't need to create the email body here, and you don't need to pass along the dummy email name etc. here. Just pass along the username as a single parameter and let send_dummy_mail_to_admin handle the other stuff.

In other words, everything else can be done in the domain/services layer -- this is not the appropriate place for it. (Controllers should be "thin".)

Reply:
Done.